mlrun/serving/states.py

```diff
@@ -110,37 +111,47 @@ def next(self):
 
     @property
     def parent(self):
-        """state parent (flow/router)"""
+        """step parent (flow/router)"""
         return self._parent
 
     def set_next(self, key: str):
-        """set/insert the key as next after this state, optionally remove other keys"""
+        """set/insert the key as next after this step, optionally remove other keys"""
         if not self.next:
             self._next = [key]
         elif key not in self.next:
             self._next.append(key)
         return self
 
-    def after_state(self, after):
-        """specify the previous state name"""
-        # most states only accept one source
+    def after_step(self, after):
+        """specify the previous step name"""
+        # most steps only accept one source
         self.after = [after] if after else []
         return self
 
-    def error_handler(self, state_name: str):
+    def after_state(self, after):
+        warnings.warn(
+            "This method is deprecated. Use after_step instead",
+            # TODO: In 0.7.0 do changes in examples & demos In 0.9.0 remove
+            PendingDeprecationWarning,
+        )
+        return self.after_step(after)
+
+    # Replaced state_name with step_name without backwards compatibility since this will usually not be used
+    # as a named parameter - step.error_handler("name")
+    def error_handler(self, step_name: str):
```

Comment:
Usually - maybe, I don't know, but let's be fully backwards compatible and support also the case if someone used it as kwarg

Reply:
The issue here is that step_name is mandatory and has no default value. This means that if someone calls this method like this: `error_handler(state_name="something")`, this will fail anyway. 
So I could either choose to give a default value to `step_name=None` or use this approach. My thinking was that in almost all cases this is called without named parameters, and if someone for some reason chose to use named params then this will fail.